co-authors-plus.php

```diff
@@ -1048,23 +1048,27 @@ function filter_wp_get_object_terms( $terms, $object_ids, $taxonomies, $args ) {
 	 * @return int Post count
 	 */
 	function filter_count_user_posts( $count, $user_id ) {
-		$user = get_userdata( $user_id );
+		$user     = get_userdata( $user_id );
+		$coauthor = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
 
-		if ( is_object( $user ) ) {
-			$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
+		// Return $count, if no coauthor exists.
+		if ( ! is_object( $coauthor ) ) {
+			return $count;
+		}
 
-			$term = $this->get_author_term( $user );
+		$term = $this->get_author_term( $coauthor );
 
-			if ( $term && ! is_wp_error( $term ) ) {
-				if ( 'guest-author' === $user->type ) {
-					// If using guest author term count, add on linked user count.
-					$count = (int) $count + $term->count;
-				} else {
-					$count = $term->count;
-				}
-			}
+		// Return combined post count, if account is linked.
+		if ( is_int( $term->count ) && strlen( $coauthor->linked_account ) > 2 ) {
```

Comment:
What cases would `$term->count` not be an integer? In cases where it would not be, do you see a better check available?

Reply:
Technically, `$term->count` should always be an integer, if that variable is available. By using `is_int( $term->count )`, I wanted to practice defensive coding, thus, only enter the statement when the variable is an integer.